### PR TITLE
doc/user: Segment guide

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

This PR is an integration guide for Segment as a webhook source. This moves, formats and expands the [current guide in the `CREATE SOURCE`](https://materialize.com/docs/sql/create-source/webhook/#connecting-with-segment) to the new **Webhook** section in **Ingest Data**

This is part of the following initiative: https://github.com/MaterializeInc/materialize/issues/21197

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by ruf-io** [comment]
@joacoc A few pieces of feedback:

1. We should redirect this page  https://materialize.com/guides/segment-to-materialize/ to your new page, I've created a tracking issue here: https://github.com/MaterializeInc/www/issues/721
2. We should give a user examples or templates for parsing three common Segment events:
    1. Page - https://segment.com/docs/connections/spec/page/
    2. Track - https://segment.com/docs/connections/spec/track/
    3. Identify - https://segment.com/docs/connections/spec/identify/ - _this one is extra tricky because it actually requires you to manually write your own "upsert" logic in SQL. @chuck-alt-delete proposed here: https://materializeinc.slack.com/archives/C05HL4U0CP8/p1689786790656969?thread_ts=1689786785.070859&cid=C05HL4U0CP8

File: doc/user/content/ingest-data/segment.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        body->>'email' AS email,
+        body->>'event' AS event,
+        body->>'messageId' AS message_id,
+        body->>'properties' AS properties,
+        body->>'timestamp' AS ts,
+        body->>'type' AS event_type,
+        body->>'userId' AS user_id
+    FROM my_segment_source
+);
+```
```
Comment (sjwiesman) [inline doc/user/content/ingest-data/segment.md:128]: It looks like you should drop the with clause.

File: doc/user/content/ingest-data/segment.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+---
+
+[Segment](https://Segment.com/) is a Customer Data Platform (CDP) that collects and routes events
+from your web and mobile app. You can use Materialize to process, sink or serve each event.
+
+This guide will walk you through the steps to ingest data from Segment to Materialize.
+
+### Pre-requisite
+
+* A Segment source.
```
Comment (bobbyiliev) [inline doc/user/content/ingest-data/segment.md:18]:
```suggestion
Ensure that you have a Segment source.
```

File: doc/user/content/ingest-data/segment.md (hunk 1 above)
Comment (joacoc) [reply doc/user/content/ingest-data/segment.md:128]: Thank you for pointing that out! @bobbyiliev we should apply the same in RudderStack

**Review by bobbyiliev — CHANGES_REQUESTED** [review]

File: doc/user/content/ingest-data/segment.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+## Step 1. (Optional) Create a Materialize Cluster
+
+If you already have a cluster for your webhook sources, you can skip this step.
+
+To create a Materialize cluster, follow the steps outlined in our
+create a [cluster guide](/sql/create-cluster),
+or create a managed cluster for all your webhooks using the following SQL statement:
+
+```sql
+CREATE CLUSTER webhooks_cluster SIZE = '3xsmall', REPLICATION FACTOR = 2;
```
Comment (bobbyiliev) [inline doc/user/content/ingest-data/segment.md:29]: We have to keep it with 1 replica due to this error:
```
ERROR:  cannot create source in cluster with more than one replica
```

```suggestion
CREATE CLUSTER webhooks_cluster SIZE = '3xsmall';
```

**Comment by joacoc** [comment]
@ruf-io updated. I added also a link to the new JSON parser. If they have another event type, pasting it there would work! Do you know if the [partial event section in the webhook source docs](https://materialize.com/docs/sql/create-source/webhook/#duplicated-and-partial-events) needs an update to fit @chuck-alt-delete case?

**Review by bobbyiliev — APPROVED** [review]
LGTM!

**Pull request merged**